contracts/AedexV2Pair.aes

```diff
@@ -306,16 +306,16 @@ contract AedexV2Pair =
                 Chain.event( Mint( Contract.address, value ))
                 Chain.event( LockLiquidity(value) )
 
-    stateful function burn_update(from: address,value: int) =
+    stateful function burn_update(value: int) =
         require(value > 0, "AedexV2Pair: INSUFFICIENT_LIQUIDITY_BURNED")
         let total_supply = state.total_supply // we are using twice: verification and update
         require(total_supply >= value , "AedexV2Pair: INSUFFICIENT_TOTAL_SUPPLY")
-        let balance = state.balance_of[ from ] // using twice: verification and update
+        let balance = state.balance_of[ Contract.address ] // using twice: verification and update
         require(balance >= value , "AedexV2Pair: INSUFFICIENT_BALANCE_FOR_BURNING")
         put(state { total_supply  = total_supply - value
-                  , balance_of[ from ] = balance - value
```

Comment:
I think i can see a theoretical case in which the `balance_of[Contracts.address]` might be modified after line 188
https://github.com/aeternity/dex-contracts-v2/blob/458967ceb2833dde3f3393b08d0ebce92325f8c9/contracts/AedexV2Pair.aes#L186-L188

The case (even if not probable or useful) but possible from the point of view of AedexV2Pair, is when `state.factory.fee_to() == Some(Contract.address)` and some extra balance will be added to balance of the contract after line 188. After this moment `balance_of != liquidity`

Reply:
Having `Contract.address` as the fee collector would be bad though since the way to burn liquidity tokens is to transfer them to `Contract.address` and then call `burn` i.e. the next caller of `burn` would then also burn the tokens collected as fee 😅